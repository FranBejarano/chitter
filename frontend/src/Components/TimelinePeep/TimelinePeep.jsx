import axios from "axios";
import React, { useEffect, useState } from "react";

import Peep from "../Peep/Peep";

const TimelinePeep = () => {
  const [timeLine, setTimeLine] = useState(null);

  useEffect(() => {
    const fetchData = async () => {
      try {
        const timelinePeeps = await axios.get(
          `/peeps/explore`
        );
        setTimeLine(timelinePeeps.data);
      } catch (err) {
        console.log("error", err);
      }
    };

    fetchData();
  }, []);

  return (
    <>

      <div className="mt-6 max-w-max">
        {timeLine &&
          timeLine.map((peep) => {
            return (
              <div key={peep._id} className="max-w-screen-md border-y-2 hover:bg-slate-100">
                <Peep peep={peep} />
              </div>
            );
          })}
      </div>

    </>
  );
};

export default TimelinePeep;
