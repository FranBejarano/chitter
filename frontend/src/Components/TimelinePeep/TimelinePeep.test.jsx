import React from "react";
import renderer from 'react-test-renderer';
import TimelinePeep from "./TimelinePeep";

jest.mock("axios");
jest.mock("../Peep/Peep");

const renderTree = tree => renderer.create(tree);
describe('<TimelinePeep>', () => {
  it('2. should render component', () => {
    expect(renderTree(<TimelinePeep
    />).toJSON()).toMatchSnapshot();
  });

});