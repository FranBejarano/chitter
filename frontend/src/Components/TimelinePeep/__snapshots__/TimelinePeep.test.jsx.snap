// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelinePeep> 2. should render component 1`] = `
<div
  className="mt-6 max-w-max"
/>
`;
